app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java

```diff
@@ -262,7 +269,17 @@ public int onStartCommand(Intent intent, int flags, int startId) {
      * function is called to notify this class to run/terminate the syncthing binary.
      * {@link #onServiceStateChange} is called while applying the decision change.
      */
-    private void onUpdatedShouldRunDecision(boolean newShouldRunDecision) {
+    private void onUpdatedShouldRunDecision(RunConditionCheckResult result) {
+        boolean newShouldRunDecision = result.isShouldRun();
+        boolean reasonsChanged;
+        synchronized (mCurrentCheckResult) {
```

Comment:
Why do you need to synchronize on an atomic? That makes no sense.

Reply:
Because I want consistency over the compare and assignment. I was thinking the same, but otherwise the assignment will race.

Reply:
Fixed by your suggestion of using `getAndSet()` instead.